rabbitmq-server: Fix building on Darwin

DaGenix opened the pull request:
On Darwin, settin HOME=/build before calling rabbitmqctl doesn't work: it produces an error while trying to create a file in /build because the /build directory doesn't exist. It also doesn't work when building using "nix develop". So, we set HOME=$TMPDIR instead which does work in those cases.


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [x] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Commit:
rabbitmq-server: Fix building on Darwin

On Darwin, settin HOME=/build before calling rabbitmqctl doesn't work:
it produces an error while trying to create a file in /build because the
/build directory doesn't exist. It also doesn't work when building using
"nix develop". So, we set HOME=$TMPDIR instead which does work in those
cases.

DaGenix commented:
@dasJ @xokdvium @samueltardieu - I think https://github.com/NixOS/nixpkgs/pull/367395 broke building on Darwin and this is my attempt to fix it. If this looks reasonable, can someone add a label to backport this to 24.11? Thanks!

DaGenix commented:
Specifically, the error that occurs is:
```
19:56:19.961 [error] Failed to create cookie file '/build/.erlang.cookie': enoent
```

samueltardieu commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 371045`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>rabbitmq-server</li>
    <li>rabbitmq-server.doc</li>
    <li>rabbitmq-server.man</li>
  </ul>
</details>

samueltardieu commented:
> It's still missing `__darwinAllowLocalNetworking = true;`

@DaGenix Can you add it?

Commit:
rabbitmq-server: Allow local networking on Darwin

This is required to allow the installCheckPhase to work on Darwin with
the sandbox on.

DaGenix commented:
@samueltardieu I made the change you requested

samueltardieu commented:
> @samueltardieu I made the change you requested

Thanks. Can you build the package with sandbox enabled?

DaGenix commented:
I tried it with sandbox enabled and it built for me.

On Sun, Jan 5, 2025, 2:05 PM Samuel Tardieu ***@***.***>
wrote:

> @samueltardieu <https://github.com/samueltardieu> I made the change you
> requested
>
> Thanks. Can you build the package with sandbox enabled?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/371045#issuecomment-2571721427>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAQA7W4RLGOA5P3K24MK2GT2JF65ZAVCNFSM6AAAAABUTVVLKOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKNZRG4ZDCNBSG4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

samueltardieu commented:
@xokdvium Would you have a chance to retry it?

xokdvium reviewed (approved):
Builds fine for me now:

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 371045`

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>rabbitmq-server</li>
    <li>rabbitmq-server.doc</li>
    <li>rabbitmq-server.man</li>
  </ul>
</details>
